.travis.yml

```diff
@@ -3,12 +3,30 @@ go:
   - 1.7
   - tip
 
+install:
+  - wget https://github.com/google/protobuf/releases/download/v3.1.0/protoc-3.1.0-linux-x86_64.zip
+  - unzip protoc-3.1.0-linux-x86_64.zip -d /tmp
+  - sudo cp /tmp/bin/protoc /usr/bin/protoc
+  - sudo chmod 777 /usr/bin/protoc
+  - rm protoc-3.1.0-linux-x86_64.zip
+
 before_script:
-  - go get github.com/ChimeraCoder/gojson/gojson
+  - go get -u github.com/ChimeraCoder/gojson/gojson
+  - go get -u github.com/golang/protobuf/protoc-gen-go
   - go generate
+  # We need to ignore changes to this one file
+  # because the go-generated version will have a different
+  # fileDescriptor0 depending on which version of Go was used
+  # to build protoc-gen-go.
+  - git checkout ssf/sample.pb.go
+
   - gofmt -w .
-  - test -n $(git status --porcelain)
 
+  # If `go generate` or `gofmt` yielded any changes,
+  # this will fail with an error message like "too many arguments"
+  # or "M: binary operator expected"
```

Comment:
Sure - dropping here for posterity:

> the problem - git does a quick mtime check as an optimization to compare files to the index quickly

> running gofmt and go generate touches everything and bumps their mtimes

> so diff-index without --cached sees that the mtimes have changed, and therefore it can't compare the files to the index unless it reads them (which it explicitly is not supposed to do)
 
> so it simply reports a bunch of zeroes, "file may have changed"
 
> running update-index --refresh bumps the times forward without changing any of the sha1s actually stored in the index

> so if you run diff-index in ANY of the containers, it always thinks the working tree is "maybe changed"

> my guess is it's an overlayfs stat peculiarity

> like, i can go all the way to the container made after `ADD . /go/src/github.com/stripe/veneur`

> and diff-index will still report dirty on all the files

> so we'll have to be sure to run update-index --refresh && diff-index HEAD in a single step

(though `git add . && git diff-index cached` accomplishes the same thing)